src/printer.js

```diff
@@ -4088,14 +4120,14 @@ function printAssignment(
         )
       ))
   ) {
-    return group(full, {shouldBreak: true})
+    return group(full, {shouldBreak: true, visibleType: 'assignment'})
```

Comment:
Introduced new `assignment` visible type so that we can parenthesize or not based on whether it breaks without affecting existing (outer) `visible` visible type logic

Eg this correctly formats like this (call parens can still be omitted based on outer call args `visible` group breaking even though inner `assignment` group doesn't break):
```
a(
  b ?= cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc dd
  e
)
```